src/analog/m2kanalogin_impl.cpp

```diff
@@ -105,55 +91,60 @@ void M2kAnalogInImpl::init()
 
 	for (unsigned int i = 0; i < getNbChannels(); i++) {
 		enableChannel(i, true);
+		auto ch = static_cast<ANALOG_IN_CHANNEL>(i);
 
-		ANALOG_IN_CHANNEL ch = static_cast<ANALOG_IN_CHANNEL>(i);
 		m_trigger->setAnalogMode(ch, ALWAYS);
-
 		setRange(ch, PLUS_MINUS_25V);
-		m_adc_calib_offset.at(i) = 2048;
-		m_adc_calib_gain.at(i) = 1;
-		m_adc_hw_vert_offset.at(i) = 0;
-		m_adc_hw_vert_offset_raw.at(i) = 2048;
+		setAdcCalibOffset(ch, 2048);
+		setAdcCalibGain(ch, 1);
+		setVerticalOffset(ch, 0);
 		m_trigger->setCalibParameters(i, getScalingFactor(i), m_adc_hw_vert_offset.at(i));
 	}
-	setKernelBuffersCount(1);
+	setKernelBuffersCount(4);
 }
 
 void M2kAnalogInImpl::syncDevice()
 {
 	for (unsigned int i = 0; i < getNbChannels(); i++) {
-		//enabled???
-		ANALOG_IN_CHANNEL ch = static_cast<ANALOG_IN_CHANNEL>(i);
-		auto range = getRangeDevice(ch);
-		m_input_range[i] = range;
+		auto range = getRangeDevice(static_cast<ANALOG_IN_CHANNEL>(i));
+		m_input_range.at(i) = range;
 
 		// load calib_offset from the register - assuming it was deinitialized correctly
-		m_adc_calib_offset[i] = m_ad5625_dev->getLongValue(2 + i, "raw", true);
-		getVerticalOffset(ch);
+		m_adc_calib_offset.at(i) = m_m2k_adc->getLongValue(i, "calibbias", false);
+		m_adc_calib_gain.at(i) = m_m2k_adc->getDoubleValue(i, "calibscale", false);
+		m_adc_hw_offset_raw.at(i) = m_ad5625_dev->getLongValue(2 + i, "raw", true);
+		m_adc_hw_vert_offset.at(i) = convertRawToVoltsVerticalOffset(static_cast<ANALOG_IN_CHANNEL>(i), m_adc_hw_offset_raw.at(i) - m_adc_calib_offset.at(i));
+
+		m_samplerate = m_m2k_adc->getLongValue("sampling_frequency");
+		m_oversampling_ratio =  m_m2k_adc->getLongValue("oversampling_ratio");
 
 		m_trigger->setCalibParameters(i, getScalingFactor(i), m_adc_hw_vert_offset.at(i));
-		m_samplerate = getSampleRate();
 	}
 }
 
 void M2kAnalogInImpl::setAdcCalibGain(ANALOG_IN_CHANNEL channel, double gain)
 {
-	m_adc_calib_gain[channel] = gain;
+
+	m_adc_calib_gain.at(channel) = m_m2k_adc->setDoubleValue(channel, gain, "calibscale");
 	m_trigger->setCalibParameters(channel, getScalingFactor(channel), m_adc_hw_vert_offset.at(channel));
 }
 
 void M2kAnalogInImpl::setAdcCalibOffset(ANALOG_IN_CHANNEL channel, int raw_offset)
 {
-	m_adc_calib_offset[channel] = raw_offset;
+	int rawVertOffset = m_adc_hw_offset_raw.at(channel) - m_adc_calib_offset.at(channel);
```

Comment:
This int can be made `const` as it is never modified.

Reply:
Done.